colexec: minor cleanup

yuzefovich opened the pull request:
**colexec: clean up ordinalityOp**

This commit changes the planning of ordinalityOp to specify the output
column index (previously, the column was simply appended on the first
call to Next). This unifies ordinalityOp with all other operators. Also,
in theory it is possible that different batches are returned by the
input to ordinalityOp, and the later batches might not have the appended
column, but I cannot reproduce it in practice (probably because
ordinality processor is only planned in non-distributed fashion, and if
several nodes stream data to it, the streams will have been merged
somehow). Anyway, I think this change is beneficial.

Release note: None

**colexec: remove redundant unset of Nulls in cfetcher**

Nulls are reset as part of ResetInternalBatch call below, so unsetting
them explicitly is redundant and is now removed.

Release note: None

**colexec: clean up any_not_null_agg template**

This commit removes explicit template language to align
any_not_null_tmpl with others.

Release note: None

yuzefovich committed:
colexec: clean up ordinalityOp

This commit changes the planning of ordinalityOp to specify the output
column index (previously, the column was simply appended on the first
call to Next). This unifies ordinalityOp with all other operators. Also,
in theory it is possible that different batches are returned by the
input to ordinalityOp, and the later batches might not have the appended
column, but I cannot reproduce it in practice (probably because
ordinality processor is only planned in non-distributed fashion, and if
several nodes stream data to it, the streams will have been merged
somehow). Anyway, I think this change is beneficial.

Release note: None

yuzefovich committed:
colexec: remove redundant unset of Nulls in cfetcher

Nulls are reset as part of ResetInternalBatch call below, so unsetting
them explicitly is redundant and is now removed.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44184)
<!-- Reviewable:end -->

yuzefovich committed:
colexec: clean up any_not_null_agg template

This commit removes explicit template language to align
any_not_null_tmpl with others.

Release note: None

asubiotto reviewed (changes requested):
> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44184)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 170 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/44184#-LzCYbEu14IxIsnVTH7l:-LzCYbEv-XWdAUwjbpU2:b-6al2gn) ([raw file](https://github.com/cockroachdb/cockroach/blob/a4ab7e13059d7980a880c056b796238b18b2ad3f/pkg/sql/colexec/any_not_null_agg_tmpl.go#L170)):*
> ```Go
> // current group, then the output for the current group is set to null.
> func _FIND_ANY_NOT_NULL(a *anyNotNull_TYPEAgg, nulls *coldata.Nulls, i int, _HAS_NULLS bool) { // */}}
> 	// {{define "findAnyNotNull" -}}
> ```

What does adding the `-` do?
___
*[pkg/sql/colexec/execgen/cmd/execgen/any_not_null_agg_gen.go, line 45 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/44184#-LzCYpgDEla6WmNVxO_h:-LzCYpgDEla6WmNVxO_i:bjkqzri) ([raw file](https://github.com/cockroachdb/cockroach/blob/a4ab7e13059d7980a880c056b796238b18b2ad3f/pkg/sql/colexec/execgen/cmd/execgen/any_not_null_agg_gen.go#L45)):*
> ```Go
> 	}
> 
> 	return tmpl.Execute(wr, sameTypeComparisonOpToOverloads[tree.EQ])
> ```

Not sure I understand this change.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44184)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/sql/colexec/any_not_null_agg_tmpl.go, line 170 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/44184#-LzCYbEu14IxIsnVTH7l:-LzD1O7TBB4RCxZlqOhI:b-xyj5z7) ([raw file](https://github.com/cockroachdb/cockroach/blob/a4ab7e13059d7980a880c056b796238b18b2ad3f/pkg/sql/colexec/any_not_null_agg_tmpl.go#L170)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

What does adding the `-` do?
</blockquote></details>

Without it, an empty line is put into the generated file, but if you add `-` to the end of a template definition it will omit it. Also, it seems to omit the next line in the tmpl file from the generated, and that's why I moved it one line above. I think I noticed it in one of Rafi's PRs and more or less learned the behavior of this trick by trial and error.
___
*[pkg/sql/colexec/execgen/cmd/execgen/any_not_null_agg_gen.go, line 45 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/44184#-LzCYpgDEla6WmNVxO_h:-LzD2NYc4lX6D-yXoIr5:bn8ow29) ([raw file](https://github.com/cockroachdb/cockroach/blob/a4ab7e13059d7980a880c056b796238b18b2ad3f/pkg/sql/colexec/execgen/cmd/execgen/any_not_null_agg_gen.go#L45)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

Not sure I understand this change.
</blockquote></details>

This change is so that we can use `.LTyp` as an argument to `replaceManipulationFuncs`. Previously, we were iterating over all coltypes, and they have methods (like `Set` and `Get`), but AFAIU we would need to call `replaceManipulationFuncs` with empty argument for type identity (i.e. `""`) and that doesn't work. This change switches to iterating over all same types comparison overloads which have the types inside of it, and we can call functions on them. Maybe there is a way to do this with iterating over `coltypes.AllTypes`, but I tried all variations I could think of, and none worked. The generated code is the same (modulo deleted comment).


<!-- Sent from Reviewable.io -->

asubiotto reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44184#-:-LzDK9eq2Lx2XGhwEuuV:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/sql/colexec/execgen/cmd/execgen/any_not_null_agg_gen.go, line 45 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/44184#-LzCYpgDEla6WmNVxO_h:-LzDK-Dm9mLA3i_XUDy3:b-8aciik) ([raw file](https://github.com/cockroachdb/cockroach/blob/a4ab7e13059d7980a880c056b796238b18b2ad3f/pkg/sql/colexec/execgen/cmd/execgen/any_not_null_agg_gen.go#L45)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

This change is so that we can use `.LTyp` as an argument to `replaceManipulationFuncs`. Previously, we were iterating over all coltypes, and they have methods (like `Set` and `Get`), but AFAIU we would need to call `replaceManipulationFuncs` with empty argument for type identity (i.e. `""`) and that doesn't work. This change switches to iterating over all same types comparison overloads which have the types inside of it, and we can call functions on them. Maybe there is a way to do this with iterating over `coltypes.AllTypes`, but I tried all variations I could think of, and none worked. The generated code is the same (modulo deleted comment).
</blockquote></details>

Yeah, I remember this being annoying.


<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
